File: contest/LeetCode_Weekly_Contest_206/Special_Positions_in_a_Binary_Matrix.java
class Solution {
    public int numSpecial(int[][] mat) {
        int n = mat.length;
        int m = mat[0].length;

        int ret = 0;
        for (int i = 0; i < n; ++i)
            for (int j = 0; j < m; ++j) if (mat[i][j] == 1) {
                boolean flag = true;
                for (int c = 0; c < m; ++c) if (c != j && mat[i][c] == 1) {
                    flag = false;
                    break;
                }
                if (!flag) continue;
                for (int r = 0; r < n; ++r) if (r != i && mat[r][j] == 1) {
                    flag = false;
                    break;
                }
                if (flag) ret++;
            }
        return ret;
    }
}

File: contest/LeetCode_Weekly_Contest_206/Special_Positions_in_a_Binary_Matrix.js
/**
 * @param {number[][]} mat
 * @return {number}
 */
var numSpecial = function(mat) {
    var n = mat.length;
    var m = mat[0].length;
    var ret = 0;
    for (var i = 0; i < n; i++)
        for (var j = 0; j < m; j++) if (mat[i][j]) {
            var flag = true;
            for (var r = 0; r < n; r++) if (r != i && mat[r][j]) {
                flag = false;
                break;
            }
            if (!flag) continue;
            for (var c = 0; c < m; c++) if (c != j && mat[i][c]) {
                flag = false;
                break;
            }
            if (flag) ret++; 
        }
    return ret;
};
